src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParser.g4

```diff
@@ -908,7 +910,9 @@ javadocInlineTag:
             | LINKPLAIN_LITERAL (WS | NEWLINE | LEADING_ASTERISK)* reference description?
             | LITERAL_LITERAL (WS | NEWLINE | LEADING_ASTERISK | text)*
             | VALUE_LITERAL (WS | NEWLINE | LEADING_ASTERISK)* reference?
-            | CUSTOM_NAME (WS | NEWLINE | LEADING_ASTERISK)* description?
+            | CUSTOM_NAME (WS | NEWLINE | LEADING_ASTERISK)*
+              ({!isNextJavadocTag()}? LEADING_ASTERISK | htmlComment
+              | CDATA | NEWLINE | text | javadocInlineTag | htmlElement)*
```

Comment:
Again, as @romani says in the issue description - we cannot know how a custom tag should behave - it might not recognize HTML or other javadoc tags inside as well as `{@code }` for example. So we should allow only text.